### DISCUSSION TITLE
How could I mock React's forwardRef?

### DISCUSSION THREAD

**Discussion by tothdanieldamit** [vitest-dev/vitest#5711] [Q&A]
All tests fails because I used a component in the tree with forwardRef..
How could i mock or make some use of it?

### Minimal reproduction code
```tsx
import {forwardRef} from 'react';

export const SpecialButton = forwardRef((prop, ref) => (
	<Button {...props} ref={ref} />
));
```

**Comment by chrgue** [comment]
Same here:
`TypeError: vi.mocked(...).mockImplementation is not a function`

Without `forwardRef` it is working.
